packages/components/src/dateInput/DateInput.js

```diff
@@ -2,58 +2,90 @@
 
 import React from 'react';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
-import { format, setMonth } from 'date-fns';
 import { View } from 'react-native';
 import {
   StyleSheet,
   Picker,
   Text,
   TextInput,
 } from '@kiwicom/universal-components';
+import { setYear, setMonth, setDate, format } from 'date-fns';
 
 type Props = {|
-  +onDateChange: (string, string) => void,
-  +date: {
-    +day: string,
-    +month: string,
-    +year: string,
-  },
-  +errors: {
-    +day: string,
-    +year: string,
-  },
+  +onDateChange: Date => void,
+  +date: ?Date,
 |};
 
 type Month = {|
   value: string,
   label: string,
 |};
 
-class DateInput extends React.Component<Props> {
-  getMonthOptions = () => {
-    return [...Array(12).keys()].map<Month>(
-      (currentMonth: number): Month => ({
-        value: String(currentMonth),
-        label: format(setMonth(new Date(), currentMonth), 'MMMM'),
-      }),
-    );
+type State = {|
+  day: string,
+  month: string,
+  year: string,
+|};
+
+const getMonthOptions = () => {
+  return [...Array(12).keys()].map<Month>(
+    (currentMonth: number): Month => ({
+      value: String(currentMonth),
+      label: format(setMonth(new Date(), currentMonth), 'MMMM'),
+    }),
+  );
+};
+
+export default class DateInput extends React.Component<Props, State> {
+  state = {
+    day: '',
+    month: '0',
+    year: '',
+  };
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (this.state !== prevState) {
+      if (Object.values(this.state).every(el => el !== '')) {
+        if (this.state.year.length === 4) {
```

Comment:
Pass prop `minDate, maxDate` and validate input according it. The same use native pickers.

Reply:
I am not sure about it, as I want to do validation outside of component in order to keep error messages (in this case alert modal) consistent

Reply:
I decided to remove all validation from the component itself, its only checks if its empty or not. Although maybe its not necessary either as if the date wouldnt be composed it would be the same error message